### DISCUSSION TITLE
Visual Absorption

### DISCUSSION THREAD

**Discussion by bytetec** [FLAMEGPU/FLAMEGPU2#1213] [Q&A] [answered]
Hi all
I have an issue with visualizations which have large environment boundaries (5000) and around 700k agents. See pictures:
![image](https://github.com/FLAMEGPU/FLAMEGPU2/assets/138150718/11399d0f-0cb3-4dee-b1c6-ec1bbef86b08)
![image](https://github.com/FLAMEGPU/FLAMEGPU2/assets/138150718/fc521da9-1d47-436e-98ae-0fa2354d42f5)
![image](https://github.com/FLAMEGPU/FLAMEGPU2/assets/138150718/2b1d6573-e61d-48f3-801d-ff965e61fc70)
It looks a bit like something is between the camara and the model. With an angular view on it I see parts of the model (picture 2).  If I go closer to the model, then the obscuration disappears (picture 3). No issue when environment boundaries is set to 4000. Does anybody have similar experience?

Code:
```
if pyflamegpu.VISUALISATION:
    # Create visualisation
    m_vis = cuda_model.getVisualisation()
    # Set the initial camera location and speed
    INIT_CAM = ENV_WIDTH / 2
    m_vis.setInitialCameraTarget(INIT_CAM, INIT_CAM, 0)
    m_vis.setInitialCameraLocation(INIT_CAM, INIT_CAM, ENV_WIDTH)
    m_vis.setCameraSpeed(0.1)
    m_vis.setSimulationSpeed(25)
    # Add "bacteria" agents to the visualisation
    bacteria_agt = m_vis.addAgent("bacteria")
    # Location variables have names "x" and "y" so will be used by default
    bacteria_agt.setModel(pyflamegpu.ICOSPHERE);
    bacteria_agt.setModelScale(1/1.0);
    # Mark the environment bounds
    pen = m_vis.newPolylineSketch(1, 1, 1, 0.2)
    pen.addVertex(0, 0, 0)
    pen.addVertex(0, ENV_WIDTH, 0)
    pen.addVertex(ENV_WIDTH, ENV_WIDTH, 0)
    pen.addVertex(ENV_WIDTH, 0, 0)
    pen.addVertex(0, 0, 0)
    # Open the visualiser window
    m_vis.activate()
```

**Comment by Robadob** [comment, marked as answer]
You may want to try increasing both the near/far clip values of the view frustum.

https://github.com/FLAMEGPU/FLAMEGPU2/blob/4e0260476353710860a4c29c7bd96da27c80aa05/include/flamegpu/visualiser/ModelVis.h#L182
